### ISSUE TITLE
Support for IBM Z missing

### ISSUE THREAD

**Issue by rdapp1** [vstinner/perf#37]
When trying to run a benchmark I noticed that the function ```collect_cpu_freq()``` in ```_collect_metadata.py``` parses the output of ```/proc/cpuinfo``` but fails on the IBM Z (mainframe, S/390).

On Linux on IBM Z the output may look like this:

```
vendor_id       : IBM/S390
# processors    : 16
[...]
processor 0: version = 00,  identification = [...]
cpu number      : 0
cpu MHz dynamic : 5208
cpu MHz static  : 5208
[...]
```

The code splits lines starting with `processor` at ':' which will not yield a proper CPU number on Z. It should suffice, however, to check the `vendor_id` for `IBM/S390` and look for `cpu number` in this case instead. The parsing of of the frequency should work ok with the existing logic.

**Comment by haypo** [comment]
Do you want to work on a patch?

**Comment by rdapp1** [comment]
No idea why the email response to this thread did not go through, here it is again:

Unfortunately, it would take me a while to obtain copyright assignment
for a new project :(  I already have a fix on my machine that is rather
trivial and works as far as I can tell, I just cannot post it here for
legal reasons. Therefore, I assume it will be faster if you fix it.

**Comment by vstinner** [comment]
Fixed by commit 7febee0242ce8fd01f56bce2ec72ec536a41caed which will be in the 1.6 release that I will release shortly.
